homeassistant/components/android_ip_webcam/binary_sensor.py

```diff
@@ -5,45 +5,50 @@
     BinarySensorDeviceClass,
     BinarySensorEntity,
 )
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import CONF_HOST, CONF_NAME, DATA_IP_WEBCAM, KEY_MAP, AndroidIPCamEntity
+from . import AndroidIPCamBaseEntity, AndroidIPCamDataUpdateCoordinator
+from .const import DOMAIN, MOTION_ACTIVE
 
 
-async def async_setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the IP Webcam binary sensors."""
-    if discovery_info is None:
-        return
+    """Set up the IP Webcam sensors from config entry."""
 
-    host = discovery_info[CONF_HOST]
-    name = discovery_info[CONF_NAME]
-    ipcam = hass.data[DATA_IP_WEBCAM][host]
+    coordinator: AndroidIPCamDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
 
-    async_add_entities([IPWebcamBinarySensor(name, host, ipcam, "motion_active")], True)
+    async_add_entities([IPWebcamBinarySensor(coordinator)])
 
 
-class IPWebcamBinarySensor(AndroidIPCamEntity, BinarySensorEntity):
+class IPWebcamBinarySensor(AndroidIPCamBaseEntity, BinarySensorEntity):
     """Representation of an IP Webcam binary sensor."""
 
     _attr_device_class = BinarySensorDeviceClass.MOTION
 
-    def __init__(self, name, host, ipcam, sensor):
+    def __init__(
+        self,
+        coordinator: AndroidIPCamDataUpdateCoordinator,
+    ) -> None:
         """Initialize the binary sensor."""
-        super().__init__(host, ipcam)
-
-        self._sensor = sensor
-        self._mapped_name = KEY_MAP.get(self._sensor, self._sensor)
-        self._attr_name = f"{name} {self._mapped_name}"
-        self._attr_is_on = None
-
-    async def async_update(self):
-        """Retrieve latest state."""
-        state, _ = self._ipcam.export_sensor(self._sensor)
-        self._attr_is_on = state == 1.0
+        super().__init__(coordinator)
+        self._attr_name = f"{coordinator.config_entry.data[CONF_NAME]} {MOTION_ACTIVE}"
```

Comment:
Consider setting `_attr_has_entity_name` for non imported entities and following the entity naming guidelines.

For imported config entries we need to keep the legacy name to avoid breaking change until we remove the import.

https://developers.home-assistant.io/docs/core/entity#entity-naming

Reply:
Here's an example of how to solve this:
https://github.com/home-assistant/core/pull/76390/files#diff-1a250a7ee76e0836bee90716f677b9bb771475687301e4eaf46030b73fb2bea3R111-R118